onset_maproom/calc.py

```diff
@@ -51,6 +51,29 @@ def water_balance(
     return water_balance
 
 
+def longest_spell(flagged_data, coord):
+    """Find the longest spell of flagged (0/1) data along coord
+    The longest spell is the maximum value
+    of the discrete difference
+    of cumulative flags
+    keeping only the unflagged data
+    Because diff needs at least 2 points,
+    we need to keep (where) the unflagged and first and last
+    with the cumulative value for last
+    and 0 for first
+    """
+    # Points to apply diff to
+    unflagged_and_ends = (flagged_data == 0) * 1
+    unflagged_and_ends[{coord: [0, -1]}] = 1
+    
+    ls = flagged_data.cumsum(coord).where(unflagged_and_ends, drop=True).where(
```

Comment:
I don't think this will work as you intended with multidimensional inputs. `where` with `drop=True` only drops a coordinate label if *all* the mask values with that label are False. That is, if the input has dimensions (T, Y, X), then T=i will be dropped only if mask[i, x, y] is False for all values of x and y.

Demonstration: if the DataArray has only one dimension, then there are no NaNs in the result of the `where`
```
data = [1, 0]
da = xr.DataArray(data=data)
print(da.where(da, drop=True))
# <xarray.DataArray (dim_0: 1)>
# array([1.])
```
but if it has two dimensions, there are NaNs in the result.
```
data = [[1, 0], [0, 1]]
da = xr.DataArray(data=data)
print(da.where(da, drop=True))
# <xarray.DataArray (dim_0: 2, dim_1: 2)>
# array([[ 1., nan],
#        [nan,  1.]])
```

Reply:
good catch